lxd/storage_pools_config.go

```diff
@@ -0,0 +1,162 @@
+package main
+
+import (
+	"fmt"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/lxc/lxd/shared"
+)
+
+var storagePoolConfigKeys = map[string]func(value string) error{
+	"source": shared.IsAny,
+	"size":   shared.IsAny,
+	"volume.block.mount_options":  shared.IsAny,
+	"volume.block.filesystem":     shared.IsAny,
+	"volume.size":                 shared.IsAny,
+	"volume.zfs.use_refquota":     shared.IsBool,
+	"volume.zfs.remove_snapshots": shared.IsBool,
+	"volume.lvm.thinpool_name":    shared.IsAny,
+	"zfs.pool_name":               shared.IsAny,
```

Comment:
That's a LOT more "IsAny" than I'd like to see here... Surely we have some way of validating byte amounts in existing code for "size" and "volume.size".

For "volume.block.filesystem", we should check for either "ext4" or "xfs" as they are the only supported ones right now.

The rest are probably fine as Any for now though I'm sure they also have some kind of length limitation and character restriction we may eventually want to check too.